assets/javascripts/modules/toggleDynamicFormFields.js

```diff
@@ -1,33 +1,39 @@
 require('jquery');
 
-module.exports = function() {
+module.exports = function () {
   var $dynamicForm = $('form[data-dynamic-form]');
-
-  $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function() {
-    if ($(this).prop('checked')) {
-      var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-      $toggledField.addClass('js-hidden');
-    }
-  });
-
-  $dynamicForm.on('click', '*[data-dynamic-fields]', function() {
-    var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-    if ($(this).data('dynamic-fields-hide')) {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', false);
+   $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function () {
+      if ($(this).prop('checked')) {
+          var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
+          $toggledField.addClass('js-hidden');
       }
-
-      $toggledField.addClass('js-hidden');
-      $toggledField.find(':text').val('');
-      $toggledField.find(':checked').prop('checked', false);
-      $toggledField.find('*[data-default]').prop('checked', true);
-    } else {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', true);
+      $('.data-' + $(this).data('nested-children')).each(function () {
```

Comment:
there is no need for the data attribute `nested-children` here. You can use the `$toggledField` variable, you will need to define it outside of the checked if conditional